src/main/java/uk/nhs/digital/nhsconnect/nhais/service/EdifactToMeshMessageService.java

```diff
@@ -22,6 +37,20 @@ public MeshMessage toMeshMessage(TranslatedInterchange translatedInterchange) {
         }
         meshMessage.setWorkflowId(WorkflowId.REGISTRATION);
         meshMessage.setContent(translatedInterchange.getEdifact());
+        meshMessage.setMessageSentTimestamp(timestampService.formatInISO(timestampService.getCurrentTimestamp()));
+        return meshMessage;
+    }
+
+    public MeshMessage fromEdifactString(String edifactString, String messageId) {
+        Interchange interchange = edifactParser.parse(edifactString);
+        //send to inbound queue
+        MeshMessage meshMessage = new MeshMessage();
+        meshMessage.setContent(edifactString);
+        meshMessage.setOdsCode(meshCypherDecoder.getSender(edifactString));
+        meshMessage.setWorkflowId(RecepMessage.isRecep(edifactString) ? WorkflowId.RECEP : WorkflowId.REGISTRATION);
```

Comment:
Same here, TranslatedInterchange contains the interchangeType == workflowFlow id.